File: Sum of Left Leaves/Sum_of_Left_Leaves.cpp
// 4ms 99.50%

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
#include<queue>
using namespace std;
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {

        if (!root)
            return 0;
        
        queue<TreeNode*> q;
        q.push(root);
        
        int sum = 0;

        while (!q.empty())
        {
            root = q.front();
            q.pop();

            if (root->left && !root->left->left && !root->left->right)
                sum += root->left->val;
            if (root->left)q.push(root->left);
            if (root->right)q.push(root->right);
        }
        return sum;
    }
};

File: Sum of Left Leaves/Sum_of_Left_Leaves.py
# 64ms 65.08%

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root is None:
            return 0
        queue = []
        queue.append(root)  
        
        sum = 0
        while len(queue) > 0:
            root = queue.pop(0)

            if root.left is not None and root.left.left is None and root.left.right is None:
                sum += root.left.val
            if root.left:
                queue.append(root.left)
            if root.right:
                queue.append(root.right)
        return sum